backend/database-router.js

```diff
@@ -1,50 +1,96 @@
 'use strict'
 
-const readConfig = () => {
-  const configString = process.env.GI_PERSIST_ROUTER_CONFIG
-  const config = JSON.parse(configString)
+import { resolve } from 'node:path'
+import { readFile } from 'node:fs/promises'
+import DatabaseBackend from './DatabaseBackend.js'
+import type { IDatabaseBackend } from './DatabaseBackend.js'
 
-  if (!config['*']) {
-    throw new Error('Fallback storage (*) is required')
+type Config = {
+  [string]: { name: string, options: Object }
+}
+type ConfigEntry = { name: string, options: Object }
+
+const { GI_PERSIST_ROUTER_CONFIG_PATH = './database-router-config.json' } = process.env
```

Comment:
@taoeffect I know this was a requested change, but I'm not happy with this being the sole option. The way it was before, which could read the config straight from an environment variable, is easier in certain situations, such as with containers.

Reply:
OK either you or @snowteamer are free to later add such an envvar

Reply:
I'm ok to implement this, but what to do if both `GI_PERSIST_ROUTER_CONFIG`  and `GI_PERSIST_ROUTER_CONFIG_PATH` are defined?
- Make one of them have priority over the other
- Raise an error because of the ambiguity

Reply:
@snowteamer I would make the envvar take precedence and add a comment to that effect